libs/ledger-live-common/src/families/stacks/bridge/utils/misc.ts

```diff
@@ -135,22 +149,28 @@ export const getAccountShape: GetAccountShape = async info => {
   const balanceResp = await fetchBalances(address);
   const rawTxs = await fetchFullTxs(address);
   const mempoolTxs = await fetchFullMempoolTxs(address);
+  const pendingTxs: MempoolTransaction[] = [];
 
   const balance = new BigNumber(balanceResp.balance);
   let spendableBalance = new BigNumber(balanceResp.balance);
   for (const tx of mempoolTxs) {
     spendableBalance = spendableBalance
       .minus(new BigNumber(tx.fee_rate))
       .minus(new BigNumber(tx.token_transfer.amount));
+
+    if (tx.tx_status === "pending") {
+      pendingTxs.push(tx);
+    }
   }
 
-  const result = {
+  const result: Partial<Account> = {
     id: accountId,
     xpub: publicKey,
     freshAddress: address,
     balance,
     spendableBalance,
     operations: flatMap(rawTxs, mapTxToOps(accountId, info)),
+    pendingOperations: flatMap(pendingTxs, mapTxToOps(accountId, info)),
```

Comment:
Sorry, I don't think it's the good approach to edit `pendingOperations` yourself, because unlike `operations` it has a fully generic lifecycle and that would interfere with it (pending operations are only added after a broadcast, and only removed by a generic mechanism to reconcile them with confirmed operations found during sync)

Do you think you could just call `fetchFullMempoolTxs` during `findNextNonce`? That would shorten the loop between when you need the info and when you get it.

Reply:
I see what you mean. I didn't know this, but I was thinking if I change between devices the pending txns are lost and now I get it, it's by design. This can be reverted quickly as we now know the problem and we fix it the way you suggest. The `fetchFullMempoolTxs` doesn't take long so can be done in `findNextNonce`.

Reply:
Fixed @haammar-ledger, please have another look 😄